tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -14,15 +14,31 @@
 #include "debug/dprint.h"
 #include "debug/assert.h"
 
+// The command queue write interface controls writes to the completion region, host owns the completion region read interface
+// Data requests from device and event states are written to the completion region
+
+CQWriteInterface cq_write_interface;
+
 constexpr uint32_t dispatch_cb_base = get_compile_time_arg_val(0);
 constexpr uint32_t dispatch_cb_log_page_size = get_compile_time_arg_val(1);
 constexpr uint32_t dispatch_cb_pages = get_compile_time_arg_val(2);
 constexpr uint32_t dispatch_cb_sem = get_compile_time_arg_val(3);
 constexpr uint32_t dispatch_cb_blocks = get_compile_time_arg_val(4);
+constexpr uint32_t completion_queue_base_addr = get_compile_time_arg_val(5);
+constexpr uint32_t completion_queue_size = get_compile_time_arg_val(6);
 
 constexpr uint32_t prefetch_noc_xy = uint32_t(NOC_XY_ENCODING(PREFETCH_NOC_X, PREFETCH_NOC_Y));
 constexpr uint32_t dispatch_noc_xy = uint32_t(NOC_XY_ENCODING(DISPATCH_NOC_X, DISPATCH_NOC_Y));
+constexpr uint64_t pcie_noc_xy_encoding = uint64_t(NOC_XY_ENCODING(PCIE_NOC_X, PCIE_NOC_Y)) << 32;
```

Comment:
Nit: I lean towards storing the noc_xy as uint32_t and using the dataflow api noc_addr_helper to construct the 32 bit version so we don't rely on magical shifts.  Since this is known at compile time, not a big deal to do it either way.  If noc addr wasn't known, unclear to me if reserving 64 bits ahead of time would burn 2 registers (vs just 1 for the 32 bit noc_xy).

Reply:
Will update